tokio/src/runtime/task/mod.rs

```diff
@@ -450,3 +472,17 @@ unsafe impl<S> linked_list::Link for Task<S> {
         NonNull::from(target.as_ref().owned.with_mut(|ptr| &mut *ptr))
     }
 }
+
+impl fmt::Display for Id {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl Id {
+    fn next() -> Self {
+        use std::sync::atomic::{AtomicUsize, Ordering::Relaxed};
+        static NEXT_ID: AtomicUsize = AtomicUsize::new(1);
+        Self(NEXT_ID.fetch_add(1, Relaxed))
```

Comment:
i would kind of prefer task IDs to always be 64-bit, but i don't love turning this into a mutex on targets without 64-bit atomics...

Reply:
I think you kind of have to, if you go sequential, always have it be u64. u32 actually could wrap.

Reply:
Okay, IDs are now always 64-bit!